### ISSUE TITLE
Bogus error message from `Pattern[str]`

### ISSUE THREAD

**Issue by rebcabin** [lcompilers/lpython#1945]
repro : https://github.com/rebcabin/lpython/tree/brian-lasr/lasr/lpython/Issue1945

The following works in ordinary CPython (after uncommenting the compatibility block at the top of the file)

```python
hex_pat : Pattern[str] = fr'-?0[xX]{hexit}+'
oct_pat : Pattern[str] = fr'-?0{octit}+'
int_pat : Pattern[str] = fr'-?{digit}+'
dec_pat : Pattern[str] = fr'(-?{digit})M'
# Not kosher Clojure
sym_pat : Pattern[str] = fr'({alpha}{asr_alpha_mer}*)'
```

LPython produces the following misleading error message:

```bash
└─(15:13:05 on brian-lasr ✭)──> ~/CLionProjects/lpython/src/bin/lpython -I. lasr_lexer.py && ./lasr_lexer.out                                             ──(Sun,Jun18)─┘
semantic error: Variable 'str' not declared
   --> lasr_lexer.py:159:19
    |
159 | hex_pat : Pattern[str] = fr'-?0[xX]{hexit}+'
    |                   ^^^ 
```

The error message does not lead me to a fix.

**Comment by Thirumalai-Shaktivel** [comment]
Is Pattern a `dataclass` or `importing type` from some other module?

**Comment by rebcabin** [comment]
`Pattern` is imported from `typing` in ordinary CPython

**Comment by certik** [comment]
LPython needs to reject all type annotations that it does not recognize. So it should say `The Pattern type is not supported.`.

**Comment by Shaikh-Ubaid** [comment]
>LPython needs to reject all type annotations that it does not recognize. So it should say The Pattern type is not supported..

I handled it here https://github.com/lcompilers/lpython/pull/1975. After merging/completion of that PR, LPython would print
```bash
$ lpython examples/expr2.py 
semantic error: Unsupported type annotation: Pattern
 --> examples/expr2.py:2:15
  |
2 |     hex_pat : Pattern[str] = fr'-?0[xX]{hexit}+'
  |               ^^^^^^^^^^^^ 


Note: if any of the above error or warning messages are not clear or are lacking
context please report it to us (we consider that a bug that must be fixed).
```

**Comment by certik** [comment]
I think this is now fixed. @rebcabin if the error message is still not clear, let's open up a new issue or reopen this one. Thanks @Shaikh-Ubaid for fixing it.
